gtsfm/common/gtsfm_data.py

```diff
@@ -387,3 +387,53 @@ def align_via_Sim3_to_poses(self, wTi_list_ref: List[Optional[Pose3]]) -> "Gtsfm
             aligned_data.add_track(track_aligned)
 
         return aligned_data
+
+    @classmethod
+    def align_data_to_axes(cls, gtsfm_data: "GtsfmData", walignedTw: Pose3) -> None:
+        """Align the GtsfmData's tracks and poses to the x,y,z axes.
+
+        Args:
+            gtsfm_data: data to perform alignment on.
+            walignedTw: transformation required to align point cloud and frustums to axes.
+
+        Raises:
+            ValueError: if walignedTw is not a valid Pose3 object.
+        """
+        if walignedTw is None:
+            raise ValueError("walignedTw cannot be None, should be a valid alignment transformation.")
+
+        num_tracks = gtsfm_data.number_tracks()
+        aligned_data = cls(gtsfm_data.number_images())
+
+        # Populate aligned_gtsfm_data with duplicate camera data.
+        for camera_index in gtsfm_data.get_valid_camera_indices():
+            aligned_data.add_camera(camera_index, gtsfm_data.get_camera(camera_index))
+
+        # Populate aligned_data with tracks.
+        for i in range(num_tracks):
+            original_track_3d = gtsfm_data.get_track(i)
+            point = original_track_3d.point3()
+            aligned_point = walignedTw.matrix() @ np.append(point, [1], 0)
```

Comment:
we can just use `aligned_point = walignedTw.transformFrom(point)` like we do in line 380.